Devnot.Mentor.Api/Repositories/Interfaces/IMenteeRepository.cs

```diff
@@ -2,17 +2,17 @@
 using DevnotMentor.Api.Entities;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using DevnotMentor.Api.CustomEntities.Request.CommonRequest;
 
 namespace DevnotMentor.Api.Repositories.Interfaces
 {
     public interface IMenteeRepository : IRepository<Mentee>
     {
+        Task<List<Mentee>> SearchAsync(SearchRequest request);
```

Comment:
Arama işlemini mentee veya mentor üzerinden yapıyoruz. Eğer buna ihtiyacımız yoksa, User üzerinden yaparsak; tek bir repository metodu, tek bir servis metodu ve tek bir end-point üzerinden menteeleri veya mentorları dönebiliriz.

Eğer buna ihtiyacımız varsa, bu şekilde kalabilir.

Reply:
Title, description gibi alanlar mentor ya da mentee ile ilgili şeyler diğer kalan alanlar da kullanıcı ile ilgili. Bu yüzden user hariç 2 repository üzerinde de koşmak gerekiyor. Generic yapı ile halledilir ama o kadar efora gerek yok diye düşündüm.

Reply:
Dediğin gibi çok daha fazla vakit ayırmamız gerekiyor, daha uygun bir zamanda düzenleriz.